Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.h

```diff
@@ -119,6 +119,18 @@ namespace AZ
         friend class Console;
     };
 
+    template<class T, class = void>
+    struct ConsoleCommandMemberFunctorSignature
```

Comment:
question: This has nothing to do with the actual changes sorry but just to check is this `class = void` usage a SFINAE trick to be used in conjunction with the definition below? I still struggle remembering how all these template tricks work 😅

Reply:
Yes, what is occurring here is that the base using the ConsoleCommandFunctorSignature template with a single parameter would first use the base template to find what the parameters should be.
i.e `ConsoleCommandMemberFunctorSignature<int>` complete type is in `ConsoleCommandMemberFunctorSignature<int, void>`.
As the template parameters were viable in the base template, specializations are then examined to see if there more specialized template. The specializations that result in an `<int, void>` parameter after argument substitution are the viable candidates. 
The specialization of struct `ConsoleCommandMemberFunctorSignature <T, AZStd::enable_if_t<AZStd::is_class_v<T>>>`, uses `std::enable_if_t` which exposes a type alias to `void`, if the condition succeeds, therefore allowing it to fulfill the requirement of `ConsoleCommandMemberFunctorSignature<T, void>` when the condition succeeds(i.e T is class).
If T is not a class, then `std::enable_if_t` does not expose a type alias, result in a substitution error removing the specialization from being a candidate.

Reply:
I see! 😅 Thank you for the detailed explanation @lumberyard-employee-dm, much appreciated (would love you to write a tutorial we could share on this at some point! 😝)